pandas/src/joins_func_helper.pxi.in

```diff
@@ -10,18 +11,25 @@ WARNING: DO NOT edit .pxi FILE directly, .pxi is generated from .pxi.in
 
 {{py:
 
-# table_type, by_dtype
-by_dtypes = [('PyObjectHashTable', 'object'), ('Int64HashTable', 'int64_t')]
+# by_dtype, table_type, init_table, s1, s2, s3, g1, g2
+by_dtypes = [('int64_t', 'Int64HashTable', 'Int64HashTable(right_size)',
+              '.set_item(', ', ', ')',
```

Comment:
why are you changing this for object table? it makes the code much more complicated. We don't use python objects anywhere in cython (instead we use the PyObjectHashTable).